src/test/java/core/basesyntax/service/RegistrationServiceImplTest.java

```
@@ -0,0 +1,162 @@
+        user.setLogin("Oleksandr123");
+        user.setPassword("1234567");
+        user.setAge(21);
+
+        User registered = registrationService.register(user);
+        assertNotNull(registered);
+        assertEquals("1234567", registered.getPassword());
+    }
+
+    @DisplayName("register() with short password should throw exception")
+    @Test
+    void register_invalidPassword_notOk() {
+        User user = new User();
+        user.setLogin("Oleksandr123");
+        user.setPassword("123");
+        user.setAge(21);
+
+        RegistrationException ex = assertThrows(RegistrationException.class,
+                () -> registrationService.register(user));
+        assertEquals("Password must be at least 6 characters", ex.getMessage());
+    }
+
+    @DisplayName("register() with valid age should work")
+    @Test
+    void register_validAge_ok() {
+        User user = new User();
+        user.setLogin("Oleksabdr123");
+        user.setPassword("1234567");
+        user.setAge(19);
+
+        User registered = registrationService.register(user);
+        assertEquals(19, registered.getAge());
+    }
+
+    @DisplayName("register() with age under 18 should throw exception")
+    @Test
+    void register_ageLessThan18_notOk() {
+        User user = new User();
+        user.setLogin("Oleksabdr123");
+        user.setPassword("1234567");
+        user.setAge(17);
+
+        RegistrationException ex = assertThrows(RegistrationException.class,
+                () -> registrationService.register(user));
+        assertEquals("User must be at least 18 years old", ex.getMessage());
+    }
+
+    @DisplayName("register() with null password should throw exception")
+    @Test
+    void register_nullPassword_notOk() {
+        User user = new User();
+        user.setLogin("validLogin");
+        user.setPassword(null);
+        user.setAge(18);
+
+        assertThrows(NullPointerException.class, () -> registrationService.register(user));
```

Issue: The test expects a NullPointerException when registering a user with a null password. According to the checklist, your service should throw a RegistrationException for invalid input, not a NullPointerException. Please update this test to expect a RegistrationException with a relevant message for null password.